### PR TITLE
pdi-scanner: TS client for pdictl

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Adds a thin TS client for `pdictl`. My goal was to have as little logic as possible in this layer and have it faithfully represent the `pdictl` API to keep things as simple as possible. The main logic here is to translate command+response combos into function calls and to register listeners for events.

## Demo Video or Screenshot
N/A

## Testing Plan
Manual testing so far. I plan to come back and add automated testing as a follow up once we feel good about this API.

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: libs/pdi-scanner/ts/src/scanner_client.ts [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+type SimpleResult = Result<void, ScannerError>;
+
+function loggableResponse(response: PdictlResponse) {
+  if (response.type === 'scanComplete') {
+    return {
+      ...response,
+      imageData: response.imageData.map((imageData) => {
+        return `${imageData.length} bytes`;
+      }),
```
Comment (eventualbuddha) [inline libs/pdi-scanner/ts/src/scanner_client.ts:122]:
```suggestion
      imageData: response.imageData.map((imageData) => `${imageData.length} bytes`),
```

File: libs/pdi-scanner/ts/src/scanner_client.ts [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+        break;
+      }
+      case 'scanComplete': {
+        emit({
+          type: 'scanComplete',
+          images: mapSheet(response.imageData, (imageData) => {
+            const buffer = Buffer.from(imageData, 'base64');
+            const grayscaleImage = createImageData(
+              Uint8ClampedArray.from(buffer),
+              1728,
```
Comment (eventualbuddha) [inline libs/pdi-scanner/ts/src/scanner_client.ts:172]: Extract to a constant?

File: libs/pdi-scanner/ts/src/scanner_client.ts [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+  // pdictl only processes one command at a time, so once we send a command, we
+  // can wait on a single response.
+  let pendingResponse: Deferred<PdictlResponse> | undefined;
+
+  // Listen for output from pdictl. We may receive either a response to a
+  // command or an unsolicited event.
+  const rl = createInterface(pdictl.stdout);
+  rl.on('line', (line) => {
+    const response = JSON.parse(line) as PdictlResponse;
+    debug('received:', loggableResponse(response));
```
Comment (eventualbuddha) [inline libs/pdi-scanner/ts/src/scanner_client.ts:158]:
```suggestion
    debug('received: %o', loggableResponse(response));
```

File: libs/pdi-scanner/ts/src/scanner_client.ts [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+ * communicates with it over stdin/stdout.
+ *
+ * A client can only be used for one lifetime of the `pdictl` process. Once the
+ * process exits and the 'disconnected' error code is returned, the client will
+ * no longer be able to send commands and a new client should be created to
+ * reconnect to the scanner.
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function createPdiScannerClient() {
+  const pdictl = spawn(PDICTL_PATH, { detached: true });
```
Comment (eventualbuddha) [inline libs/pdi-scanner/ts/src/scanner_client.ts:139]: Document why `detached: true`?

**Review by eventualbuddha — APPROVED** [review]

File: libs/pdi-scanner/ts/src/scanner_client.ts (hunk 4 above)
Comment (jonahkagan) [reply libs/pdi-scanner/ts/src/scanner_client.ts:137]: This was necessary to allow me to intercept SIGINT when I was trying to make the demo script close the scanner connection gracefully. Now that we're not worried about that anymore, it's not necessary. Removing

**Pull request merged**